tests/test_embedding_caching.py

```diff
@@ -34,30 +39,72 @@ def cache_dir(self, tmp_path):
     def test_caching_functionality(self, cache_dir):
         # Create a dummy model
         dummy_model = DummyModel()
+        dummy_task_metadata_query = TaskMetadata(
+            name="DummyTaskQuery",
+            description="Dummy task metadata",
+            dataset={"path": "test", "revision": "test"},
+            type="Classification",
+            eval_langs=["eng-Latn"],
+            main_score="accuracy",
+        )
+        dummy_task_metadata_corpus = dummy_task_metadata_query.copy()
+        dummy_task_metadata_corpus.name = "DummyTaskCorpus"
 
         # Create the wrapper
         wrapped_model = CachedEmbeddingWrapper(dummy_model, cache_dir)
 
         # Simulate data
-        queries = [
-            "What is the effect of vitamin C on common cold?",
-            "How does exercise affect cardiovascular health?",
-        ]
-        corpus = [
-            "Vitamin C supplementation does not significantly reduce the incidence of common cold.",
-            "Regular exercise improves cardiovascular health by strengthening the heart and reducing blood pressure.",
-            "The impact of vitamin C on common cold duration is minimal according to recent studies.",
-        ]
+        queries = DataLoader(
```

Comment:
Do we have a test with images to be sure? I'm not sure the easiest way to do that though